airflow/sensors/base.py

```diff
@@ -182,7 +184,11 @@ def __init__(
         self.mode = mode
         self.exponential_backoff = exponential_backoff
         self.max_wait = self._coerce_max_wait(max_wait)
+        if soft_fail is True and never_fail is True:
+            raise ValueError("soft_fail and never_fail are mutually exclusive, you can not provide both.")
```

Comment:
I think that it is important to clarify the mutual exclusivness also in the docstrings (both in `soft_fail` and `never_fail`)

Reply:
yes , it's done